### DISCUSSION TITLE
Suggestions for Improvement of ScreenSleep

### DISCUSSION THREAD

**Discussion by RadicalRadish** [adrianlungu/ScreenSleep#8] [Ideas]
Hello Adrian,

Good little app that adds useful functionality. However, I have a couple of suggestions for improvement.

(1) As it stands when the screen is sleeping ScreenSleep monitors the mouse for (a) any click and/or (b) mouse move on the desk. Would suggest that there should be setting such that the user can turn off monitoring the mouse when the monitor is sleeping. Couple of reasons why for this. (A) As the screen is sleeping that user can't see what the rest of the system is doing. This means that, for example, the system might have put up some kind of dialogue box that does as certain action if the mouse if clicked - this could lead to the user having the computer 'do something' without being aware that they have just done that. (B) I just think it is not a good idea to wake the monitor on a **mouse move** -- Windows doesn't do this on a full Windows (system) sleep so I think that ScreenSleep should act in the same way. In all, I think that waking monitor should only happen on a keyboard press. So in all I would think that at least the user should be able to select if they want ScreenSleep to monitor the mouse or the keyboard or both while the monitor is sleeping.

(2) Make ScreenSleep operate in such a way that it can be hidden from the System Tray. And, to 'unhide' the user needs to (re-)launch ScreenSleep. On such a launch ScreenSleep can see that it is already running and, because this is so, it unhides/reveals itself in the System Tray (thus allowing the user access to alter settings). Once the user has altered settings then a right-click on the ScreenSleep tray icon should allow the user to hide the icon from the tray again.

Hope my ideas are thought okay,

Radish

**Comment by adrianlungu** [comment]
Hello @RadicalRadish 

Thanks for your suggestions.

I'll be looking into the 1st one real soon and it might come in the next version, the 2nd one is also a good idea and I'll look into that one as time permits.
